cogeo_mosaic/backends/auto.py

```diff
@@ -0,0 +1,38 @@
+"""
+cogeo_mosaic.backend.auto: Automatically select backend from url
+
+This is in a separate module to prevent circular imports
+"""
+
+from urllib.parse import urlparse
+
+from cogeo_mosaic.backends.dynamodb import DynamoDBBackend
+from cogeo_mosaic.backends.file import FileBackend
+from cogeo_mosaic.backends.http import HttpBackend
+from cogeo_mosaic.backends.s3 import S3Backend
+
+
+def auto_backend(url: str, **kwargs):
```

Comment:
I'm not 💯 sure about the `name` auto_backend but can't come up with a better name. If we assume this will be the most used method maybe something like `MosaicReader`? 

```python
with MosaicReader(url) as mosaic:
    asset = mosaic.tile(x, y, z) 
```

Reply:
Well clearly I'm bad at naming 😅. @geospatial-jeff ?

Reply:
Renamed to `MosaicBackend`